scripts/xtensa-build-zephyr.py

```diff
@@ -618,13 +564,11 @@ def main():
 	else:
 		print("Building platforms: {}".format(" ".join(args.platforms)))
 
-	# we made two modes mutually exclusive with argparse
-	if args.west_path:
-		# check west_path input + create symlink if needed
-		run_point_mode()
-	if args.clone_mode:
+	west_init()
```

Comment:
No. this is a check whether west tool is initialized. If it is not initialized we need to initialize it.
If we won't initialize it we can't use it.

Reply:
When starting from scratch the sequence is
1. west init
2. west update
3. west build

Why would anyone want to do 1 and 3 and skip 2? It's strange and not intuitive.

If the workspace is not initialized then the script should just fail with an error message saying "use --(init-and)-update".

When starting not from scratch it's either 2 and 3; or just 3.
1 and 3 is never useful.

Reply:
> Why would anyone want to do 1 and 3 and skip 2? It's strange and not intuitive.

Because a developer has checked out his custom branch on sof or zephyr when working on something?
`west init` we take out from equation because it simply always needs to happen otherwise this script is useless.
So you question is basically - why would anyone use west build without west update and I think my answer above makes sense.

> So I tried to invoke the script while "forgetting" the --update option. It did west init without west update which ends up in this confusing initialized-without-git-repos state and this confusing error message:

I'll provide exception handling here to explain that Zephyr is missing (where `west build` custom command is defined).